File: BasicCalc.py
class Solution:
    def calculate(self, s: str) -> int:
        s = s.strip()
        st = []
        curr = 0
        lastExp = "+"
        res = 0
        n = len(s)

        for i in range(n):
            if s[i].isnumeric():
                curr = 10 * curr + (ord(s[i]) - ord('0'))
            
            if (not s[i].isnumeric() and s[i] != " ") or i == n - 1:
                if lastExp == "+":
                    st.append(curr)
                elif lastExp == "-":
                    st.append(-1 * curr)
                elif lastExp == "*":
                    st.append(st.pop() * curr)
                elif lastExp == "/":
                    st.append(int(st.pop() / curr))
                
                lastExp = s[i]
                curr = 0

        while st:
            res += st.pop()

        return res

File: basicCalc.cpp
class Solution {
public:
    int calculate(string s) {
        deque<int> st; 
        char lastExp;
        int result = 0; 
        int currNum = 0;

        for(int i = 0; i <= s.size(); i++) {
            if(s[i] == ' ') continue; 
            if(i != s.size() && isdigit(s[i])) {
                currNum = currNum * 10 + (s[i] - '0');
            } else {
                int prevNum; 
                switch(lastExp) {
                    case '+' : 
                        st.push_back(currNum);
                        break; 
                    case '-':
                        st.push_back(currNum * -1);
                        break;
                    case '*':
                        prevNum = st.back(); 
                        st.pop_back(); 
                        st.push_back(prevNum*currNum);
                        break; 
                    case '/':
                        prevNum = st.back(); 
                        st.pop_back(); 
                        st.push_back(prevNum/currNum);
                        break; 
                    default: 
                        st.push_back(currNum);
                        break; 
                }
                lastExp = s[i];
                currNum = 0; 
            }
        }

        while(!st.empty()) {
            result += st.back(); 
            st.pop_back(); 
        }         

        return result; 
    }
};